File: clients/csharp/src/Actions/UpdateObjectRequest.cs
using System.Text.Json.Serialization;
using MyMind.Models;

namespace MyMind.Actions;

public class UpdateObjectRequest
{
    [JsonPropertyName("title")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public string? Title { get; init; }

    [JsonPropertyName("tags")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public List<ObjectTag>? Tags { get; init; }

    [JsonPropertyName("spaces")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public List<ObjectSpace>? Spaces { get; init; }
}

File: clients/swift/src/Actions/UpdateObjectRequest.swift
import Foundation

public struct UpdateObjectRequest: Codable, Sendable {
    public var title: String?
    public var tags: [ObjectTag]?
    public var spaces: [ObjectSpace]?

    public init(title: String? = nil, tags: [ObjectTag]? = nil, spaces: [ObjectSpace]? = nil) {
        self.title = title; self.tags = tags; self.spaces = spaces
    }
}
